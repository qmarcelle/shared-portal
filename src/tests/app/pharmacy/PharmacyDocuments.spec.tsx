import PharmacyPage from '@/app/pharmacy/page';
import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { mockedAxios } from '../_mocks_/axios';

const renderUI = async () => {
  const page = await PharmacyPage();
  return render(page);
};

const vRules = {
  user: {
    vRules: { medicare: true, dsnpGrpInd: false, showPharmacyTab: true },
  },
};

jest.mock('src/auth', () => ({
  auth: jest.fn(),
}));

describe('PharmacyDocuments', () => {
  it('should render UI correctly - for Medicare Groups', async () => {
    const mockAuth = jest.requireMock('src/auth').auth;
    mockAuth.mockResolvedValueOnce(vRules);
    mockedAxios.get.mockResolvedValueOnce({
      data: {},
    });
    const component = await renderUI();

    expect(screen.getByText('View Covered Drug List (Formulary)'));
    expect(
      screen.getByText('Download a list of all the drugs your plan covers.'),
    );

    expect(screen.getByText('Prescription Drug Claim Form'));
    expect(
      screen.getByText(
        // eslint-disable-next-line quotes
        "We call it the G0-510 form. You'll use it only for prescription drug claims when the claim is not filled through your pharmacist.",
      ),
    );

    expect(screen.getByText('Prescription Mail Service Order Form'));
    expect(
      screen.getByText(
        'Mail order is easy and convenient. You can have your prescriptions delivered right to your home.',
      ),
    );

    expect(
      screen.getByText(
        'Request for Medicare Prescription Drug Coverage Determination',
      ),
    );
    expect(
      screen.getByText(
        'You can request an exception for prescription drug coverage.',
      ),
    );

    expect(
      screen.getByText(
        'Request for Redetermination of Medicare Prescription Drug Denial',
      ),
    );
    expect(
      screen.getByText(
        'If you received a Notice of Denial of Medicare Prescription Drug Coverage, you can ask us for a redetermination (appeal).',
      ),
    );

    expect(component.baseElement).toMatchSnapshot();
  });

  it('should render UI correctly - for Non Medicare Groups', async () => {
    vRules.user.vRules.medicare = false;
    const mockAuth = jest.requireMock('src/auth').auth;
    mockAuth.mockResolvedValueOnce(vRules);
    mockedAxios.get.mockResolvedValueOnce({
      data: {},
    });
    const component = await renderUI();

    expect(screen.getByText('View Covered Drug List (Formulary)'));
    expect(
      screen.getByText('Download a list of all the drugs your plan covers.'),
    );

    expect(screen.getByText('Prescription Drug Claim Form'));
    expect(
      screen.getByText(
        // eslint-disable-next-line quotes
        "We call it the G0-510 form. You'll use it only for prescription drug claims when the claim is not filled through your pharmacist.",
      ),
    );

    expect(screen.getByText('Prescription Mail Service Order Form'));
    expect(
      screen.getByText(
        'Mail order is easy and convenient. You can have your prescriptions delivered right to your home.',
      ),
    );

    expect(
      screen.queryByText(
        'Request for Medicare Prescription Drug Coverage Determination',
      ),
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText(
        'You can request an exception for prescription drug coverage.',
      ),
    ).not.toBeInTheDocument();

    expect(
      screen.queryByText(
        'Request for Redetermination of Medicare Prescription Drug Denial',
      ),
    ).not.toBeInTheDocument();
    expect(
      screen.queryByText(
        'If you received a Notice of Denial of Medicare Prescription Drug Coverage, you can ask us for a redetermination (appeal).',
      ),
    ).not.toBeInTheDocument();

    expect(component.baseElement).toMatchSnapshot();
  });
});
