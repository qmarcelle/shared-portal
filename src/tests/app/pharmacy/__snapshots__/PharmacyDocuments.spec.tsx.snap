// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PharmacyDocuments should render UI correctly 1`] = `
<body>
  <div>
    <div
      class="flex flex-col"
    >
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-col items-stretch"
        >
          <a
            href="http://www.bcbst.com/docs/pharmacy/go-510.pdf"
            style="display: inline;"
          >
            <button
              class="flex flex-row link-container text-left pl-0"
              style="max-width: max-content; height: auto; display: inline;"
              tabindex="0"
            >
              <p
                class="link"
                style="display: inline;"
              >
                View Covered Drug List (Formulary)
              </p>
              <p
                class="ml-1"
                style="display: inline;"
              >
                <img
                  alt="download Icon"
                  class="inline"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </p>
            </button>
          </a>
          <p
            class="body-1"
          >
            Download a list of all the drugs your plan covers.
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
          <div
            aria-label="divider"
            class=""
            style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
        </div>
      </div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-col items-stretch"
        >
          <a
            href="http://www.bcbst.com/docs/pharmacy/go-510.pdf"
            style="display: inline;"
          >
            <button
              class="flex flex-row link-container text-left pl-0"
              style="max-width: max-content; height: auto; display: inline;"
              tabindex="0"
            >
              <p
                class="link"
                style="display: inline;"
              >
                Prescription Drug Claim Form
              </p>
              <p
                class="ml-1"
                style="display: inline;"
              >
                <img
                  alt="download Icon"
                  class="inline"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </p>
            </button>
          </a>
          <p
            class="body-1"
          >
            We call it the G0-510 form. You'll use it only for prescription drug claims when the claim is not filled through your pharmacist.
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
          <div
            aria-label="divider"
            class=""
            style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
        </div>
      </div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-col items-stretch"
        >
          <a
            style="display: inline;"
          >
            <button
              class="flex flex-row link-container text-left pl-0"
              style="max-width: max-content; height: auto; display: inline;"
              tabindex="0"
            >
              <p
                class="link"
                style="display: inline;"
              >
                Prescription Mail Service Order Form
              </p>
              <p
                class="ml-1"
                style="display: inline;"
              >
                <img
                  alt="download Icon"
                  class="inline"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </p>
            </button>
          </a>
          <p
            class="body-1"
          >
            Mail order is easy and convenient. You can have your prescriptions delivered right to your home.
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
          <div
            aria-label="divider"
            class=""
            style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
        </div>
      </div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-col items-stretch"
        >
          <a
            style="display: inline;"
          >
            <button
              class="flex flex-row link-container text-left pl-0"
              style="max-width: max-content; height: auto; display: inline;"
              tabindex="0"
            >
              <p
                class="link"
                style="display: inline;"
              >
                Request for Medicare Prescription Drug Coverage Determination
              </p>
              <p
                class="ml-1"
                style="display: inline;"
              >
                <img
                  alt="right arrow Icon"
                  class="inline"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </p>
            </button>
          </a>
          <p
            class="body-1"
          >
            You can request an exception for prescription drug coverage.
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
          <div
            aria-label="divider"
            class=""
            style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
        </div>
      </div>
      <div
        class="flex flex-col"
      >
        <div
          class="flex flex-col items-stretch"
        >
          <a
            style="display: inline;"
          >
            <button
              class="flex flex-row link-container text-left pl-0"
              style="max-width: max-content; height: auto; display: inline;"
              tabindex="0"
            >
              <p
                class="link"
                style="display: inline;"
              >
                Request for Redetermination of Medicare Prescription Drug Denial
              </p>
              <p
                class="ml-1"
                style="display: inline;"
              >
                <img
                  alt="right arrow Icon"
                  class="inline"
                  data-nimg="1"
                  decoding="async"
                  height="40"
                  loading="lazy"
                  src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                  srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                  style="color: transparent;"
                  width="40"
                />
              </p>
            </button>
          </a>
          <p
            class="body-1"
          >
            If you received a Notice of Denial of Medicare Prescription Drug Coverage, you can ask us for a redetermination (appeal).
          </p>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
